packit_service/service/api/runs.py

```diff
@@ -83,29 +85,37 @@ def process_runs(runs):
             response_dict["trigger"] = get_project_info_from_build(srpm_build)
 
         for model_type, Model, packit_ids in (
-            ("copr", CoprBuildTargetModel, pipeline.copr_build_id),
-            ("koji", KojiBuildTargetModel, pipeline.koji_build_id),
-            ("test_run", TFTTestRunTargetModel, pipeline.test_run_id),
+            ("copr", CoprBuildGroupModel, pipeline.copr_build_group_id),
+            ("koji", KojiBuildGroupModel, pipeline.koji_build_group_id),
+            ("test_run", TFTTestRunGroupModel, pipeline.test_run_group_id),
         ):
             for packit_id in set(flatten_and_remove_none(packit_ids)):
-                row = Model.get_by_id(packit_id)
-                if row.status == BuildStatus.waiting_for_srpm:
-                    continue
-                response_dict[model_type].append(
-                    {
-                        "packit_id": packit_id,
-                        "target": row.target,
-                        "status": row.status,
-                    }
+                group_row = Model.get_by_id(packit_id)
+                target_models = (
+                    group_row.grouped_targets
+                    if isinstance(group_row, GroupModel)
+                    else [group_row]
```

Comment:
what else could it be?

Reply:
Right, this can now be removed. I added this check because I was doing one type of model at a time, so previously it was a mix of group and target models.